src/Nethermind/Nethermind.Core/ProductInfo.cs

```diff
@@ -35,11 +35,43 @@ static ProductInfo()
             Version = Version[..Math.Min(index + 9, Version.Length - 1)];
         }
 
-        ClientId = $"{Name}/v{Version}/{OS.ToLowerInvariant()}-{OSArchitecture}/dotnet{Runtime[5..]}";
+        ClientIdParts =
+        [
+            Name,
+            $"v{Version}",
+            $"{OS.ToLowerInvariant()}-{OSArchitecture}",
+            $"dotnet{Runtime[5..]}"
+        ];
+
+        ClientId = string.Join("/", ClientIdParts);
     }
 
     public static DateTimeOffset BuildTimestamp { get; }
 
+    public static string FormatClientId(string? hiddenPartsName = null)
```

Comment:
Instead of passing everywhere `hiddenPartsName` and calling this, better to configure it once here and cache it in some `PublicClientId` property.

Reply:
I've initialized the PublicClientId in the network initialization, so no need to pass any value anymore. Furthermore the PublicClientId is initialized to the ClientId by default.
But i'm not sure if it's the best place to initialize it as i don't have a good grasp of the full initialization process. So is it ok or is there another place to initialize it in order to avoid having some discrepancy between the usage of the PublicClientId ?